tree/dataframe/inc/ROOT/RDFNodes.hxx

```diff
@@ -540,6 +586,33 @@ public:
       fHasRun = true;
    }
 
+   std::shared_ptr<RDFGraphDrawing::GraphNode> GetGraph()
+   {
+      auto prevNode = fPrevData.GetGraph();
+      auto prevColumns = prevNode->GetDefinedColumns();
+
+      // Action nodes do not need to ask an helper to create the graph nodes. They are never common nodes between
+      // multiple branches
+      auto thisNode = std::make_shared< RDFGraphDrawing::GraphNode>(fHelper.GetActionName());
+      auto evaluatedNode = thisNode;
+      for (auto &column : fCustomColumns.GetColumns()) {
+         /* Each column that this node has but the previous hadn't has been defined in between,
+          * so it has to be built and appended. */
+         if (RDFGraphDrawing::CheckIfDefaultOrDSColumn(column.first, column.second))
+            continue;
+         if (std::find(prevColumns.begin(), prevColumns.end(), column.first) == prevColumns.end()) {
+            auto defineNode = RDFGraphDrawing::CreateDefineNode(column.first, column.second.get());
+            evaluatedNode->SetPrevNode(defineNode);
+            evaluatedNode = defineNode;
+         }
+      }
+
+      thisNode->AddDefinedColumns(fCustomColumns.GetNames());
+      thisNode->SetAction(HasRun());
+      evaluatedNode->SetPrevNode(prevNode);
+      return thisNode;
+   }
+
```

Comment:
I do not understand what is the fate of "evaluatedNode". Perhaps this can be clarified in a comment?